types/xdg_shell/wlr_xdg_toplevel.c

```diff
@@ -581,15 +581,15 @@ uint32_t wlr_xdg_toplevel_set_maximized(struct wlr_xdg_surface *surface,
 	assert(surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL);
 	surface->toplevel->server_pending.maximized = maximized;
 
-	return schedule_xdg_surface_configure(surface);
+	return wlr_xdg_surface_schedule_configure(surface);
 }
 
 uint32_t wlr_xdg_toplevel_set_fullscreen(struct wlr_xdg_surface *surface,
```

Comment:
This function is what **the compositor** calls when it wants to fullscreen a client, not what the client calls.

Reply:
That's the point, isn't it?

I'm not trying to make this work with _no_ intervention from the compositor, I want wlroots to send a configure event when the compositor _does_ call this function to set the client fullscreen state, whether or not the client has actually changed state.

The issue before wasn't just that sway ignored the maximize request, it was that there was no way to implement the handler using the usual api (this function) that gave the correct Wayland behavior (send a configure regardless of state change). Now compositors using the usual api can get the correct behavior, and the compositor ignoring the request is never correct.

Reply:
Oh, I see, my bad.